### ISSUE TITLE
Using `with-open` blows up

### ISSUE THREAD

**Issue by Folcon** [HumbleUI/HumbleUI#66]
```clojure
(ui/default-theme
  {}
  (with-open [purple (paint/fill 0xFF9F2B68)]
    (ui/rect purple
      (ui/label (str "Apple")))))
```

Blows up with:
```shell
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x0000000152a3c309, pid=70244, tid=775
#
# JRE version: OpenJDK Runtime Environment Homebrew (19.0.1) (build 19.0.1)
# Java VM: OpenJDK 64-Bit Server VM Homebrew (19.0.1, mixed mode, sharing, tiered, compressed oops, compressed class ptrs, g1 gc, bsd-amd64)
# Problematic frame:
# C  [libskija.dylib+0x2f0309]  _ZNK7SkPaint13nothingToDrawEv+0x9
#
# No core dump will be written. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /Users/folcon/Code/pankration/hs_err_pid70244.log
#
# If you would like to submit a bug report, please visit:
#   https://github.com/Homebrew/homebrew-core/issues
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#
```

However, the below does not:
```clojure
(ui/default-theme
  {}
  (ui/rect (paint/fill 0xFF9F2B68)
    (ui/label (str "Apple"))))
```

Both return `io.github.humbleui.ui.dynamic.Contextual` and `io.github.humbleui.ui.rect.Rect`:
```clojure
;; From the first form
=> #object[io.github.humbleui.ui.dynamic.Contextual 0x19813f6 "io.github.humbleui.ui.dynamic.Contextual@19813f6"]

;; From the second form
=> #object[io.github.humbleui.ui.dynamic.Contextual 0x2aff357a "io.github.humbleui.ui.dynamic.Contextual@2aff357a"]
```

This could just be because `io.github.humbleui.skija.Paint` is closed with the `with-open` version when it comes time to render.

I'm only mentioning it as it's [something I picked up from your canvas example](https://github.com/HumbleUI/HumbleUI/blob/main/dev/examples/canvas.clj#L45), it might be worthwhile changing that to use colours set using `with-context` higher up the app so reusing this sort of logic doesn't break? Unless there's another reason for it?

**Comment by tonsky** [comment]
Well, UI is defined as an object, on which measure/draw/event methods will later be called. That means they will need paint to remain open.

Canvas example is different, in a sense that you define draw function and paint is only used inside the paint, so it’s ok to create/destroy paints on each draw.

Memory-wide, paints are relatively lightweight objects, so it’s ok to not close them explicitly. They’ll eventually be garbage collected

**Comment by Folcon** [comment]
Ok, I might stick to just defining them using `with-context` for now then. Having stuff close like that seems to be less great.

Not sure if it's the best thing for the app to crash with `_ZNK7SkPaint13nothingToDrawEv` as the error, but you might feel like it's out of scope to handle this stuff.

I'll close this =)...

**Comment by tonsky** [comment]
Yeah, the error message is confusing. I’ll think what can be done
